### PR TITLE
Add Github set-status and add-comment tasks

### PR DESCRIPTION
**Pull request by chmouel**
Add a github directory with for now only two tasks (more to follow)  allowing to do
different github actions with the API.

I have made those tasks in _pure_ python to be able to use a minimal image and
not have to build one with `hub` or other tools.

The `github-set-status` task allows external services to mark GtiHUB commits
with an `error`, `failure`, `pending`, or `success` state, which is then
reflected in pull requests involving those commits.

The `github-set-status` task allows external services to mark GtiHUB commits
with an `error`, `failure`, `pending`, or `success` state, which is then
reflected in pull requests involving those commits.

The `github-add-comment` task let you add a comment to a pull request or an
xissue.

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Yaml file complies with [yamllint](https://github.com/adrienverge/yamllint) rules.

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: github/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+with a GitHUB token in the key `secretToken`, you can easily create it on the
+command line :
+
+```
+kubectl create secret generic github --from-literal token="MY_TOKEN"
+```
+
+## Set Status on a Commit/PR
+
+The `github-set-status` task allows external services to mark GtiHUB commits
```
Comment (sbwsg) [inline github/README.md]: nit: GtiHUB -> GitHub

File: github/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+with an `error`, `failure`, `pending`, or `success` state, which is then
+reflected in pull requests involving those commits.
+
+Statuses include as well a `description` and a `target_url`, to give the user
+infomrations about the CI statuses or a direct link to the full log.
```
Comment (sbwsg) [inline github/README.md]: nit: infomrations -> information

File: github/set_status.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        data = {
+            "state": "$(inputs.params.STATE)",
+            "target_url": "$(inputs.params.TARGET_URL)",
+            "description": "$(inputs.params.DESCRIPTION)",
+            "context": "$(inputs.params.CONTEXT)"
+        }
+        print("Sending this data to GitHUB: ")
+        print(data)
+
+        conn = http.client.HTTPSConnection("api.github.com")
```
Comment (sbwsg) [inline github/set_status.yaml]: would it be worth making this url configurable to support github enterprise?  or would there be more other things we'd need to change to make that work?

Comment (chmouel) [reply github/set_status.yaml]: i am adding as param but tbh i have no idea if there is an extra thing to do...

**Comment by chmouel** [comment]
Fyi @sbwsg I have added another task add_comment, (when you don't want to use the pullReq ressource)

**Pull request merged**